squarelet/organizations/views.py

```diff
@@ -119,37 +122,75 @@ def get_context_data(self, **kwargs):
 
     def handle_join(self, request):
         self.organization = self.get_object()
-        is_member = self.organization.has_member(self.request.user)
-        if not is_member:
-            if self.request.user.can_auto_join(self.organization):
-                # Auto-join the user to the organization (no invitation needed)
-                self.organization.memberships.create(user=self.request.user)
-                if self.organization.plan and self.organization.plan.wix:
-                    sync_wix.delay(
-                        self.organization.pk,
-                        self.organization.plan.pk,
-                        self.request.user.pk,
-                    )
-                messages.success(
-                    request, _("You have successfully joined the organization!")
-                )
-            else:
-                invitation = self.organization.invitations.create(
-                    email=request.user.email, user=request.user, request=True
-                )
-                messages.success(
-                    request,
-                    _(
-                        "Request to join the organization sent!<br><br>"
-                        "We strongly recommend reaching out directly to one or all of "
-                        "the admins listed below to ensure your request is approved "
-                        "quickly. If all of the admins shown below have left the "
-                        "organization, please "
-                        '<a href="mailto:info@muckrock.com">contact support</a> '
-                        "for assistance.<br><br>"
-                    ),
+        user = request.user
+
+        # Already a member
+        if self.organization.has_member(user):
+            return
+
+        # Auto join if allowed
+        if user.can_auto_join(self.organization):
+            self.organization.memberships.create(user=user)
+            if self.organization.plan and self.organization.plan.wix:
+                sync_wix.delay(
+                    self.organization.pk,
+                    self.organization.plan.pk,
+                    user.pk,
                 )
-                invitation.send()
+            messages.success(request, "You have successfully joined the organization!")
+            return
+
+        # For normal requests to join, check rate limit
```

Comment:
This method is getting a little long, I think factoring out the rate limiting logic into its own method would be helpful.  Would also make it easier to reuse in the future if we needed to rate limit other actions.